pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -611,13 +612,30 @@ func getVirtualDiskUUID(newDevice types.BaseVirtualDevice) (string, error) {
 
 	if b, ok := vd.Backing.(*types.VirtualDiskFlatVer2BackingInfo); ok {
 		uuidWithNoHypens := strings.Replace(b.Uuid, "-", "", -1)
-		return uuidWithNoHypens, nil
+		uuidLowerCase := strings.ToLower(uuidWithNoHypens)
+		return uuidLowerCase, nil
 	}
 	return "", ErrNoDiskUUIDFound
 }
 
+func getVirtualDiskID(volPath string, vmDevices object.VirtualDeviceList) (string, error) {
+	for _, device := range vmDevices {
+		d := device.GetVirtualDevice()
+
+		if vmDevices.TypeName(device) == "VirtualDisk" {
+			if b, ok := d.Backing.(*types.VirtualDiskFlatVer2BackingInfo); ok && b.Parent != nil {
```

Comment:
It would be helpful for those not familiar w/ the vsphere API to say exactly what is being tested here.